### DISCUSSION TITLE
Run two instances of PictureFrame, one for portrait and one for landscape pictures on both HDMI outputs of RPI4

### DISCUSSION THREAD

**Discussion by ScherfP** [pi3d/pi3d_demos#39] [Q&A]
I came along your project through the website of Wolfgang and succesfully built my own picture frame running on a Raspberry Pi 4. I wonder if it is possible to use both HDMI outputs and run two instances of PictureFrame2020. One with a config file configured with the folder for landscape photos with the ouput on monitor 1 which is in landscape mode and the other config file refering to the folder with portrait photos with output on monitor 2 which is in portrait mode.

**Comment by paddywwoof** [comment]
I think it probably is, though the load on the processor might reduce the smoothness of the fade if the images are large. You would probably run two instances of PictureFrame in one you would set the Display x and y to 0, 0 (and w,h to match the first display) in the second instance you would set x equal to the width of the first display and w,h matching the second one.

If you have a go and need more answers let me know.

**Reply by ScherfP** [reply]
Thanks for the hint. I'll give it a try and report back.

**Reply by paddywwoof** [reply]
To run the two versions of PictureFrame you will need to pass the x,y,w,h config values on the command line rather than setting them in the files. Alternatively you would need to copy and rename one version - but that would get messy.

Paddy

**Comment by ScherfP** [comment]
Sorry for coming back that late. I had the aspiration and the ambition to solve it on my own but didn't succeed.

When I first attached the second monitor the screen was simply stretched over both monitors. I tried with the suggested parameters (x,y,w,h options) and was able to start and align xinit only on the first monitor. But when I tried to start another xinit process, I was told that Display0 is already in use and another process of xinit cannot be started.

I then tried to split up the monitors to two different screens each (including the rotation for the portrait mode), in order to be able to start one xinit process on each of the monitors, via a configuration file that I placed in /usr/share/X11/xorg.conf.d/

The strange thing now is, that the settings I made in the configuration file are all fine when I'm starting the desktop via "startx" (1st monitor = landscape is left and 2nd monitor = portrait is on the right), but when I start the picture frame via xinit, I always get a mirrored view and I'm not able to start anything on the second monitor. I tried again with the x,y,w,h options or "xinit ... -- :1 ..." but I'm somehow not able to address each screen / monitor separatly. It seems like the layout section in the config file is ignored when using xinit.

**Reply by paddywwoof** [reply]
Hi, it sounds like you were making some progress. I think startx is a kind of wrapper with different settings that then runs xinit. It is possible to start picture frame from the command line with startx. But I can't remember if that works from systemd (which I assume is what you're aiming at).
 
One thing you could try is to make a parent program (could be bash but I would use python) that spawns the two pictureframe programs using subprocessing. I haven't tried this so don't know what the issues would be.
